fuzz/lnwire/replychannelrangezlib/reply_channel_range_zlib.go

```diff
@@ -0,0 +1,60 @@
+// +build gofuzz
+
+package replychannelrangezlib
+
+import (
+	"bytes"
+	"compress/zlib"
+	"encoding/binary"
+
+	fuzz "github.com/lightningnetwork/lnd/fuzz/lnwire"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// Fuzz is used by go-fuzz.
+func Fuzz(data []byte) int {
+
+	var buf bytes.Buffer
+	zlibWriter := zlib.NewWriter(&buf)
+	_, err := zlibWriter.Write(data)
```

Comment:
So we're inputting random data rather than well formed (and sorted) short channel IDs? Won't this always trigger our checks to ensure that the input data is properly sorted? Or will the fuzzer eventually incorporate these constraints into the input data it generates?

Reply:
So it *should* be the case that the fuzzer learns... checking [here](https://crypt-iq.github.io/lnwire-coverage-with-zlib.html#file25) (messed up the coverage format a bit but look for decodeShortChanIDs) you can see by hovering the mouse that `if cid.ToUint64() <= lastChanID.ToUint64() {` has 15557 hits while the error message `return 0, nil, ErrUnsortedSIDs{lastChanID, cid}` has 3 hits so it is passing the checks

Reply:
Fuzzers mainly get confused with more complicated structuring or with any sort of auth check.